test/Validate.Layout.cpp

```diff
@@ -0,0 +1,199 @@
+  "%vfunct = OpTypeFunction %voidt",
+  "%viifunct =  OpTypeFunction %voidt %intt %intt",
+  // TODO(umar): OpConstant fails because the type is not defined
+  // TODO(umar): OpGroupMemberDecorate
+  "OpLine %str 3 4",
+  "%func   = OpFunction %voidt None %vfunct",
+  "OpFunctionEnd",
+  "%func2   = OpFunction %voidt None %viifunct",
+  "%funcp1 = OpFunctionParameter %intt",
+  "%funcp2 = OpFunctionParameter %intt",
+  "OpReturn",
+  "OpFunctionEnd"
+};
+
+INSTANTIATE_TEST_CASE_P(InstructionsOrder,
+    ValidateLayout,
+    ::testing::Combine(::testing::Range((int)0, (int)instructions.size()),
+    //                                   || Instruction              || Line(s) where the instruction is valid
+    ::testing::Values(   make_pair( string("OpCapability"),             Equals<0>)
+                        , make_pair(string("OpExtension"),              Equals<1>)
+                        , make_pair(string("OpExtInstImport"),          Equals<2>)
+                        , make_pair(string("OpMemoryModel"),            Equals<3>)
+                        , make_pair(string("OpEntryPoint"),             Equals<4>)
+                        , make_pair(string("OpExecutionMode"),          Equals<5>)
+                        , make_pair(string("OpSource "),                Range<6, 9>)
+                        , make_pair(string("OpSourceContinued "),       Range<6, 9>)
+                        , make_pair(string("OpSourceExtension "),       Range<6, 9>)
+                        , make_pair(string("OpString "),                Range<6, 9>)
+                        , make_pair(string("OpName "),                  Range<10, 11>)
+                        , make_pair(string("OpMemberName "),            Range<10, 11>)
+                        , make_pair(string("OpDecorate "),              Range<12, 15>)
+                        , make_pair(string("OpMemberDecorate "),        Range<12, 15>)
+                        , make_pair(string("OpGroupDecorate "),         Range<12, 15>)
+                        , make_pair(string("OpDecorationGroup"),        Range<12, 15>)
+                        , make_pair(string("OpTypeBool"),               Range<16, 26>)
+                        , make_pair(string("OpTypeVoid"),               Range<16, 26>)
+                        , make_pair(string("OpTypeInt"),                Range<16, 26>)
+                        , make_pair(string("OpTypeVector %intt 4"),     Range<16, 26>)
+                        , make_pair(string("OpTypeMatrix %vec4 4"),     Range<16, 26>)
+                        , make_pair(string("OpTypeStruct"),             Range<16, 26>)
+                        , make_pair(string("%vfunct = OpTypeFunction"), Range<16, 26>)
+                        )));
+// clang-format on
+
+// Creates a new vector which removes the string if the substr is found in the
+// instructions vector and reinserts it in the location specified by order.
+// NOTE: This will not work correctly if there are two instances of substr in
```

Comment:
This NOTE is ok.  Alternately you could just say that we do the special thing with the *first* string that has the given substring.
Coder's choice.